compiler/rustc_ty_utils/src/ty.rs

```diff
@@ -131,7 +131,7 @@ fn param_env(tcx: TyCtxt<'_>, def_id: DefId) -> ty::ParamEnv<'_> {
     if let Some(ImplTraitInTraitData::Trait { fn_def_id, .. })
     | Some(ImplTraitInTraitData::Impl { fn_def_id, .. }) = tcx.opt_rpitit_info(def_id)
     {
-        predicates = tcx.predicates_of(fn_def_id).instantiate_identity(tcx).predicates;
+        predicates.extend(tcx.predicates_of(fn_def_id).instantiate_identity(tcx).predicates);
```

Comment:
What needs to be fixed is what I don't understand. I can add a FIXME -Zflag but unsure what else to comment because I'm not sure what needs fixing here.